File: jar-common/src/main/java/com/beancounter/common/contracts/AssetResponse.java
package com.beancounter.common.contracts;

import com.beancounter.common.model.Asset;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AssetResponse implements Payload<Asset> {
  private Asset data;
}

File: jar-contracts/src/main/kotlin/com/beancounter/common/contracts/AssetResponse.kt
package com.beancounter.common.contracts

import com.beancounter.common.model.Asset
import com.beancounter.common.model.MarketData

/**
 * Response to an asset request.
 * Optionally includes current price data when requested via includePrice query parameter.
 */
data class AssetResponse(
    override val data: Asset,
    val price: MarketData? = null
) : Payload<Asset?>